WordPress/WordPressTest/NotificationFilteringServiceTests.swift

```diff
@@ -53,73 +53,44 @@ final class NotificationFilteringServiceTests: XCTestCase {
         XCTAssertTrue(sut.wordPressNotificationsEnabled)
     }
 
-    // MARK: - Disable WordPress notifications
+    // MARK: - Should disable WordPress notifications
 
-    func testDisableWPNotificationsIfNeededInJetpackWhenFeatureFlagEnabled() {
-        setup(allowDisablingWPNotifications: true, isWordPress: false)
-        XCTAssertTrue(sut.wordPressNotificationsEnabled)
-
-        sut.disableWordPressNotificationsIfNeeded()
-
-        XCTAssertFalse(sut.wordPressNotificationsEnabled)
-    }
-
-    func testDisableWPNotificationsIfNeededInJetpackWhenFeatureFlagDisabled() {
-        setup(allowDisablingWPNotifications: false, isWordPress: false)
-        XCTAssertTrue(sut.wordPressNotificationsEnabled)
-
-        sut.disableWordPressNotificationsIfNeeded()
-
-        XCTAssertTrue(sut.wordPressNotificationsEnabled)
-    }
-
-    func testDisableWPNotificationsIfNeededInWordPresskWhenFeatureFlagEnabled() {
-        setup(allowDisablingWPNotifications: true, isWordPress: true)
-        XCTAssertTrue(sut.wordPressNotificationsEnabled)
-
-        sut.disableWordPressNotificationsIfNeeded()
-
-        XCTAssertTrue(sut.wordPressNotificationsEnabled)
-    }
-
-    // MARK: - Should filter WordPress notifications
-
-    func testShouldFilterWordPressNotificationsInWordPressWhenFeatureFlagEnabledAndWordPressNotificationsDisabled() {
+    func testShouldDisableWordPressNotificationsInWordPressWhenFeatureFlagEnabledAndWordPressNotificationsDisabled() {
         setup(allowDisablingWPNotifications: true, isWordPress: true)
         sut.wordPressNotificationsEnabled = false
 
-        XCTAssertTrue(sut.shouldFilterWordPressNotifications())
+        XCTAssertTrue(sut.shouldDisableWordPressNotifications())
     }
 
-    func testShouldFilterWordPressNotificationsInWordPressWhenFeatureFlagDisabledAndWordPressNotificationsDisabled() {
+    func testShouldDisableWordPressNotificationsInWordPressWhenFeatureFlagDisabledAndWordPressNotificationsDisabled() {
         setup(allowDisablingWPNotifications: false, isWordPress: true)
         sut.wordPressNotificationsEnabled = false
 
-        XCTAssertFalse(sut.shouldFilterWordPressNotifications())
+        XCTAssertFalse(sut.shouldDisableWordPressNotifications())
     }
 
-    func testShouldFilterWordPressNotificationsInWordPressWhenFeatureFlagEnabledAndWordPressNotificationsEnabled() {
+    func testShouldDisableWordPressNotificationsInWordPressWhenFeatureFlagEnabledAndWordPressNotificationsEnabled() {
```

Comment:
```suggestion
    func testShouldAllowNotificationsInWordPressWhenFeatureFlagEnabledAndWordPressNotificationsEnabled() {
```

If the feature flag is enabled but WP notifications are still enabled, shouldn't this be verifying that notifications are in fact enabled? 

The assertion a few lines below seems to be correct, but the name of the test here seems incorrect.

Reply:
The setter for `wordPressNotificationsEnabled` could be modified to only set a new value if the feature flag is enabled. That way, the test's name could be simplified further to `testShouldAllowNotificationsInWordPressWhenWordPressNotificationsEnabled`.

Reply:
Okay, I think I understand the confusion now.

**In my test names, I describe:**
- What I am testing
- The conditions

**I don't describe:**
- The expected results

So all the test names simply repeat the method name that I'm testing: test**ShouldDisableWordPressNotifications**.

As I understand, I am not following the expected testing naming convention for `WordPress-iOS`